internal/server/configuration/configuration_manager_local.go

```diff
@@ -26,7 +27,7 @@ func NewFileConfigurationManager(path string) Manager {
 }
 
 // ReadConfiguration returns a reader for the configuration file.
-func (cm *FileConfigurationManager) ReadConfiguration() (io.ReadCloser, error) {
+func (cm *FileConfigurationManager) ReadConfiguration(_ context.Context) (io.ReadCloser, error) {
```

Comment:
you can check if err := ctx.Err here, so if context will be cancelled at the moment it will interrupt execution